qancode.py

```diff
@@ -0,0 +1,2366 @@
+            pass
+
+    def _try_perform_click_path(self):
+        if self.click_path is not None:
+            print('Performing click path: {}'.format(self.click_path.__name__))
+            return self.click_path(self.driver)
+
+    def _expand_document_details(self):
+        expand_buttons = self.driver.wait.until(EC.presence_of_all_elements_located(
+            (By.CLASS_NAME, DocumentPreview.document_expand_buttons_class)))
+        for button in expand_buttons:
+            try:
+                button.click()
+            except:
+                pass
+
+    def _get_rid_of_test_warning_banner(self):
+        testing_warning_banner_button = WebDriverWait(self.driver, 3).until(EC.element_to_be_clickable(
+            (By.CSS_SELECTOR, NavBar.testing_warning_banner_button_css)))
+        testing_warning_banner_button.click()
+
+    @abstractmethod
+    def get_data(self):
+        pass
+
+
+class BrowserComparison(metaclass=ABCMeta):
+    """
+    ABC for comparing data between browsers.
+    """
+
+    def __init__(self, user, url, item_type, browsers, all_data):
+        self.all_data = all_data
+        self.user = user
+        self.url = url
+        self.item_type = item_type
+        self.browsers = browsers
+        self.url_data = [d for d in all_data if ((d['user'] == user)
+                                                 and (d['item_type'] == item_type)
+                                                 and (d['url'] == url))]
+
+    @abstractmethod
+    def compare_data(self):
+        pass
+
+
+class URLComparison(metaclass=ABCMeta):
+    """
+    ABC for comparing data between prod and RC given browser and user.
+    """
+
+    def __init__(self, browser, user, prod_url, rc_url, item_type, all_data, click_path=None):
+        self.browser = browser
+        self.user = user
+        self.all_data = all_data
+        self.prod_url = prod_url
+        self.rc_url = rc_url
+        self.item_type = item_type
+        self.click_path = click_path
+        self.prod_data = [d['data'] for d in all_data
+                          if ((d['url'] == prod_url)
+                              and (d['user'] == user)
+                              and (d['browser'] == browser)
+                              and (d['item_type'] == item_type)
+                              and (d['click_path'] == click_path))]
+        self.rc_data = [d['data'] for d in all_data
+                        if ((d['url'] == rc_url)
+                            and (d['user'] == user)
+                            and (d['browser'] == browser)
+                            and (d['item_type'] == item_type)
+                            and (d['click_path'] == click_path))]
+        assert len(self.prod_data) == len(self.rc_data)
+
+    @abstractmethod
+    def compare_data(self):
+        pass
+
+#########################################
+# Selenium setup and sign-in procedures. #
+#########################################
+
+
+class NewDriver(object):
```

Comment:
- Wanted to avoid confusion with little-d _driver_ and also make it somewhat explicit that this creates a new driver instead of reusing old one. In any case changed.

- Removed all explicit inheritances from object.